lib/code_writer.rb

```diff
@@ -107,6 +107,40 @@ def close
     @out.close
   end
 
+  def write_init
+    raise NotImplementedError
```

Comment:
FYI some people [get mad](http://chrisstump.online/2016/03/23/stop-abusing-notimplementederror/) about using `NotImplementedError` for this purpose, because it’s [intended](https://rubydoc.info/stdlib/core/NotImplementedError) for “when a feature is not implemented on the current platform” (e.g. if you try to call `Process.fork` [on Windows](https://ruby-doc.org/core-3.0.2/Process.html#method-c-fork)). It doesn’t matter here but it’s worth at least being aware of this opinion when you’re writing production code. (Something like `raise 'unimplemented'` should be completely uncontroversial.)